src/write.rs

```diff
@@ -151,31 +146,6 @@ impl<W: Write> Write for BzEncoder<W> {
     }
 }
 
-#[cfg(feature = "tokio")]
-impl<W: AsyncWrite> AsyncWrite for BzEncoder<W> {
-    fn shutdown(&mut self) -> Poll<(), io::Error> {
-        try_nb!(self.try_finish());
-        self.get_mut().shutdown()
-    }
-}
-
-impl<W: Read + Write> Read for BzEncoder<W> {
-    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
-        self.get_mut().read(buf)
-    }
-}
-
-#[cfg(feature = "tokio")]
-impl<W: AsyncRead + AsyncWrite> AsyncRead for BzEncoder<W> {}
-
-impl<W: Write> Drop for BzEncoder<W> {
```

Comment:
Hi @folkertdev @alexcrichton,  is there any reason to remove this?
I found that in version 0.5.0, dropping the `BzEncoder` no longer auto finishes it.